File: web_vue.client/types/viewmodel/ResponseViewModel.ts
// !! 此檔案由 generate-models.mjs 自動生成，請勿手動修改 !!
// 來源：Web_Vue.Server/ViewModels
import type { ResultTypeValue } from '../enum'

/** 統一 API 回應模型 */
export interface ResponseViewModel<T = unknown> {
  /** 回應結果狀態 */
  Status: ResultTypeValue
  /** 回應資料內容 */
  Data: T | null
  /** 回應訊息 */
  Message: string
  /** 錯誤清單 */
  ErrorList: ResponseErrorDataViewModel[]
}

/** 回應錯誤資料模型 */
export interface ResponseErrorDataViewModel {
  /** 鍵 */
  Key: string
  /** 錯誤文字清單 */
  ErrorTextList: string[]
}

File: Web_Vue.Server/ViewModels/ResponseViewModel.cs
namespace Web_Vue.Server.ViewModels;

/// <summary> 統一 API 回應模型 </summary>
public class ResponseViewModel<T> where T : class
{
    /// <summary> 回應結果狀態 </summary>
    public ResultType Status { get; set; }

    /// <summary> 回應資料內容 </summary>
    public T? Data { get; set; }

    /// <summary> 回應訊息 </summary>
    public string Message { get; set; } = string.Empty;

    /// <summary> 錯誤清單 </summary>
    public List<ResponseErrorDataViewModel> ErrorList { get; set; } = new();
}

/// <summary> 回應錯誤資料模型 </summary>
public class ResponseErrorDataViewModel
{
    /// <summary> 鍵 </summary>
    public string Key { get; set; } = string.Empty;

    /// <summary> 錯誤文字清單 </summary>
    public List<string> ErrorTextList { get; set; } = new();
}
